Specifications/QIR/Callables.md

```diff
@@ -3,6 +3,12 @@
 We use to term _callable_ to mean a subroutine in the source language.
 Different source languages use different names for this concept.
 
+### Runtime Failure
+
+There are several error conditions that are specified as causing a runtime failure.
+The `quantum__rt__fail` function is the mechanism to use to cause a runtime failure;
```

Comment:
Reading through what failures we are talking about, that seems reasonable. At some point in the future, there may be a need for having a "fail" instruction on the target machine in order for it to give more meaningful information about the state at failure, but that seems to be a different issue than the failures needed here.